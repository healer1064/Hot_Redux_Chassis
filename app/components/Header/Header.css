.root {
  background: #333;
  padding: 0 2rem;
}

.navigation {
  height: 60px;
  line-height: 60px;
}

.navigation li {
  display: inline;

  a {
    padding: 0 20px;
    color: #ddd;
    display: inline-block;
    text-decoration: none;

    &:hover, &.active {
      background: #222;
    }
  }
}
