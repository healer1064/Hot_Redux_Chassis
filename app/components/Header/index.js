// @flow

import styles from './Header.css';
import React, { Component } from 'react';
import { Link, IndexLink } from 'react-router';
import { Modal } from 'react-overlays';
import cx from 'classnames';
import HamburgerButton from 'app/components/HamburgerButton';
import Icon from 'app/components/Icon';
import Search from 'app/components/Search';

const navigationItems = [
  {
    to: '/',
    label: 'Home',
    LinkComponent: IndexLink
  },
  {
    to: '/about',
    label: 'About',
    LinkComponent: Link
  },
  {
    to: '/contact',
    label: 'Contact',
    LinkComponent: Link
  }
];

/**
 * A responsive header component with titles and menus.
 * On small screens the menu collapses into an animated hamburger menu.
 */
class Header extends Component {
  state = {
    menuOpen: false
  };

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.header}>
          <div className={styles.mobileMenu}>
            <HamburgerButton
              onClick={() => this.setState({ menuOpen: !this.state.menuOpen })}
              open={this.state.menuOpen}
            />
            <Link to="/" className={styles.mobileTitle}>
              React
            </Link>
          </div>

          <Link to="/" className={styles.desktopTitle}>
            hanse
          </Link>
          <ul
            className={cx(
              styles.navigationItems,
              this.state.menuOpen && styles.openMenu
            )}
          >
            {navigationItems.map(({ LinkComponent, to, label }) =>
              <li key={label}>
                <LinkComponent
                  to={to}
                  activeClassName={styles.activeItem}
                  onClick={() => this.setState({ menuOpen: false })}
                >
                  {label}
                </LinkComponent>
              </li>
            )}
          </ul>

          <button
            className={styles.searchButton}
            onClick={this.props.toggleSearch}
          >
            <Icon name="search" />
          </button>
        </div>

        <Modal
          show={this.props.searchOpen}
          onHide={this.props.toggleSearch}
          backdropClassName={styles.backdrop}
          backdrop
        >
          <Search
            query={this.props.location.query.q}
            onClose={this.props.toggleSearch}
          />
        </Modal>
      </div>
    );
  }
}

export default Header;
