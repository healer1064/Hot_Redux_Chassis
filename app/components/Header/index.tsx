import styles from './Header.css';
import React, { useState } from 'react';
import { NavLink, Link, NavLinkProps } from 'react-router-dom';
import Modal from 'react-modal';
import cx from 'classnames';
import qs from 'query-string';
import HamburgerButton from 'app/components/HamburgerButton';
import Icon from 'app/components/Icon';
import Search from 'app/components/Search';

const ExactLink = (props: NavLinkProps) => <NavLink {...props} exact />;

const navigationItems = [
  {
    to: '/',
    label: 'Home',
    LinkComponent: ExactLink
  },
  {
    to: '/about',
    label: 'About',
    LinkComponent: NavLink
  },
  {
    to: '/contact',
    label: 'Contact',
    LinkComponent: NavLink
  }
];

type Props = {
  openSearch: () => void;
  closeSearch: () => void;
  searchOpen: boolean;
  location: {
    search: string;
  };
};

/**
 * A responsive header component with titles and menus.
 * On small screens the menu collapses into an animated hamburger menu.
 */

function Header(props: Props) {
  const [menuOpen, setMenuOpen] = useState(false);

  const toggleMenu = () => setMenuOpen(open => !open);
  const closeMenu = () => setMenuOpen(false);
  const openSearch = () => {
    setMenuOpen(false);
    props.openSearch();
  };

  return (
    <div className={styles.root}>
      <div className={styles.header}>
        <div className={styles.mobileMenu}>
          <HamburgerButton onClick={toggleMenu} open={menuOpen} />
          <Link to="/" className={styles.mobileTitle}>
            hanse
          </Link>
        </div>

        <Link to="/" className={styles.desktopTitle}>
          hanse
        </Link>
        <ul className={cx(styles.navigationItems, menuOpen && styles.openMenu)}>
          {navigationItems.map(({ LinkComponent, to, label }) => (
            <li key={label}>
              <LinkComponent
                to={to}
                activeClassName={styles.activeItem}
                onClick={closeMenu}
              >
                {label}
              </LinkComponent>
            </li>
          ))}
        </ul>

        <button className={styles.searchButton} onClick={openSearch}>
          <Icon name="search" />
        </button>
      </div>

      <Modal
        isOpen={props.searchOpen}
        onRequestClose={props.closeSearch}
        className={styles.modalContent}
        overlayClassName={styles.backdrop}
        contentLabel="Search"
      >
        <Search
          query={qs.parse(props.location.search).q as string | undefined}
          onClose={props.closeSearch}
        />
      </Modal>
    </div>
  );
}

export default Header;
