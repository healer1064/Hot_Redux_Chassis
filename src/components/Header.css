.Header {
  background: #333;
  padding: 0 2rem;
}

.Navigation {
  height: 60px;
  line-height: 60px;
}

.Navigation ul li {
  display: inline;

  a {
    padding: 0 20px;
    color: #ddd;
    display: inline-block;
    text-decoration: none;

    &:hover, &.active {
      background: #222;
    }
  }
}
