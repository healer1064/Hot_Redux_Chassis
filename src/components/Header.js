import './Header.css';
import React from 'react';
import { Link } from 'react-router';

const Navigation = ({ items }) => (
  <nav className='Navigation'>
    <ul>{items.map(([href, label]) => <li><Link to={href}>{label}</Link></li>)}</ul>
  </nav>
);

export default () => (
  <div className='Header'>
    <Navigation items={[
      ['', 'Hjem'],
      ['', 'About'],
      ['', 'Hello']
    ]} />
  </div>
);
