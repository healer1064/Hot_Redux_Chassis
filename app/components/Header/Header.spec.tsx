import React from 'react';
import { MemoryRouter } from 'react-router';
import Header from './';
import { testComponentSnapshots } from 'tests/helpers';

const Component = (props: any) => (
  <MemoryRouter>
    <Header {...props} />
  </MemoryRouter>
);

const defaultProps = {
  location: {
    search: ''
  },
  closeSearch: () => {},
  openSearch: () => {},
  searchOpen: false
};

describe('<Header />', () => {
  testComponentSnapshots(Component, [defaultProps]);
});
