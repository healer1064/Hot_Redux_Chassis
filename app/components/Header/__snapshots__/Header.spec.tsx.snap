// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  className="root"
>
  <div
    className="header"
  >
    <div
      className="mobileMenu"
    >
      <button
        className="hamburgerButton"
        onClick={[Function]}
      >
        <div />
        <div />
        <div />
        <div />
      </button>
      <a
        className="mobileTitle"
        href="/"
        onClick={[Function]}
      >
        hanse
      </a>
    </div>
    <a
      className="desktopTitle"
      href="/"
      onClick={[Function]}
    >
      hanse
    </a>
    <ul
      className="navigationItems"
    >
      <li>
        <a
          aria-current="page"
          className="activeItem"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
    <button
      className="searchButton"
      onClick={[Function]}
    >
      <i
        className="icon ion-search"
      />
    </button>
  </div>
</div>
`;
